### PR TITLE
Release - Willow 4 and Swift 4

### PR DESCRIPTION
**Pull request by cnoon**
This PR switches Willow over to Swift 4 and, adds a Package file for SPM, and cleans up all sorts of project issues and tiny code fixes. Once this is merged, we'll be able to release Willow 4.0.0.

### CODE REVIEW

File: Example/iOS Example/Base.lproj/LaunchScreen.xib [hunk 1]
```diff
@@ -12,20 +16,20 @@
             <rect key="frame" x="0.0" y="0.0" width="480" height="480"/>
             <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
             <subviews>
-                <label opaque="NO" clipsSubviews="YES" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="  Copyright (c) 2015 Nike. All rights reserved." textAlignment="center" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" minimumFontSize="9" translatesAutoresizingMaskIntoConstraints="NO" id="8ie-xW-0ye">
+                <label opaque="NO" clipsSubviews="YES" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="  Copyright (c) 2017 Nike. All rights reserved." textAlignment="center" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" minimumFontSize="9" translatesAutoresizingMaskIntoConstraints="NO" id="8ie-xW-0ye">
```
Comment (cnoon) [inline Example/iOS Example/Base.lproj/LaunchScreen.xib:19]: All that I purposefully changed here is the date from `2015` to `2017`.

File: Example/iOS Example/Info.plist [hunk 2]
```diff
@@ -33,6 +33,7 @@
 		<string>UIInterfaceOrientationPortrait</string>
 		<string>UIInterfaceOrientationLandscapeLeft</string>
 		<string>UIInterfaceOrientationLandscapeRight</string>
+		<string>UIInterfaceOrientationPortraitUpsideDown</string>
```
Comment (cnoon) [inline Example/iOS Example/Info.plist:36]: Fixes a warning when running on device.

File: Package.swift [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import PackageDescription
+
+let package = Package(name: "Willow", swiftLanguageVersions: [4], exclude: ["Tests"])
```
Comment (cnoon) [inline Package.swift:27]: 👏🏻👏🏻👏🏻

File: Example/iOS Example/Info.plist (hunk 2 above)
Comment (ejensen) [reply Example/iOS Example/Info.plist:36]: That's odd. Normally `PortraitUpsideDown` is only suggested on iPads.

**Review by ejensen — APPROVED** [review]

**Review by EricAppel — APPROVED** [review]

**Pull request merged**